api/src/main/java/io/opentelemetry/common/Main.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+import io.opentelemetry.common.CleanReadableAttributes.TypedAttributeConsumer;
+import java.util.List;
+
+@SuppressWarnings({
+  "BadImport",
+  "UnusedVariable",
+  "PrivateConstructorForUtilityClass",
+  "MultipleTopLevelClasses",
+  "JavadocMethod"
+})
+public class Main {
+
+  public static void main(String[] args) {
+    CleanAttributes attributes =
+        CleanAttributes.newBuilder()
+            .setString("string", "I'm a String")
+            .setBoolean("boolean", true)
+            .setDouble("double", 33.444d)
+            .setLong("long", 34333L)
+            .setStringArray("stringArray", "one", "two", "three")
+            .setBooleanArray("booleanArray", true, false, true)
+            .setLongArray("longArray", 33L, 55L, 99L)
+            .setDoubleArray("doubleArray", 123.33, 6655.33, 339393.33, 3434.33)
+            .build();
+
+    System.out.println("Processing with casts:");
+    process(attributes);
+    System.out.println();
+    System.out.println("Processing with types:");
+    processTyped(attributes);
+  }
+
+  private static void processTyped(CleanAttributes attributes) {
+    attributes.forEach(
+        new TypedAttributeConsumer() {
```

Comment:
I like this a bit better than the switch. No noise (`asString`, etc) just pure consumption

Reply:
me, too, although implementing a 8-method interface is not super ergonomic, either. Especially in the cases where you don't care about the types, and just want to gather raw data [if this is a use-case?]

Reply:
Easy :) Copy all attributes to another `Attributes` before you add your own attributes to it :)

My problem is the opposite: I cannot imagine when do you to iterate over all attributes in a type-safe manner.

Reply:
any exporter needs to do this. See all the exporters in the repo that have to do this work.

Reply:
@jkwatson Maybe we should knock out `Attributes.toBuilder` now since looks like @iNikem really wants it :P

Reply:
I think that's great to have, no matter what, so yes, definitely.